src/main/java/HeadersRestTemplateService.java

```diff
@@ -0,0 +1,28 @@
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpHeaders;
+import org.springframework.stereotype.Service;
+
+@Service
+public class HeadersRestTemplateService {
+
+    @Value("${github.token}")
+    private String githubToken;
+
+    @Value("${openai.api.key}")
+    private String openaiApiKey;
+
+    public HttpHeaders createGitHubHeaders() {
+        HttpHeaders headers = new HttpHeaders();
+        headers.set("Authorization", "Bearer " + githubToken);
+        headers.set("Content-Type", "application/json");
+        return headers;
+    }
+
+    public HttpHeaders createOpenAiHeaders() {
+        HttpHeaders headers = new HttpHeaders();
+        headers.set("Authorization", "Bearer " + openaiApiKey);
+        headers.set("Content-Type", "application/json");
+        return headers;
+    }
```

Comment:
The methods createGitHubHeaders and createOpenAiHeaders have a lot of duplicated code. Both methods create an HttpHeaders object and set the same Content-Type header. This violates the DRY (Don't Repeat Yourself) principle. (Severity: medium)
**Suggestion**: Consider creating a private helper method that accepts the token as a parameter and sets the headers accordingly.